# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/WaylandBuffer.h
# Commit: 9b286c5c32d1
# Full Hash: 9b286c5c32d1accc0025872508b34a1eeb0d2d64
# Author: stransky <stransky@redhat.com>
# Date: 2025-07-22 04:26:57
# Regressor Bug: 1978207
# File Overlap Count: 4
# Description:
#   Bug 1978207 [Wayland] Remove detached WaylandBuffer transactions from WaylandSurface r=emilio
#   
#   When WaylandBuffer transaction is detached from compositor, remove it from WaylandSurface.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D257976
# ==============================================================================

diff -r d73d4fbe1330 -r 9b286c5c32d1 widget/gtk/WaylandBuffer.h
--- a/widget/gtk/WaylandBuffer.h	Mon Jul 21 15:15:52 2025 +0000
+++ b/widget/gtk/WaylandBuffer.h	Mon Jul 21 15:19:40 2025 +0000
@@ -201,9 +201,14 @@
            mBufferState == BufferState::WaitingForDelete;
   }
   bool IsDetached() { return mBufferState == BufferState::Detached; }
+  bool IsDeleted() { return mBufferState == BufferState::Deleted; }
 
   void DeleteTransactionLocked(const WaylandSurfaceLock& aSurfaceLock);
 
+  bool MatchesBuffer(uintptr_t aBuffer) {
+    return aBuffer == reinterpret_cast<uintptr_t>(mBuffer.get());
+  }
+
  private:
   ~BufferTransaction();
 